packages/emotion-server/src/index.js

```diff
@@ -15,12 +15,10 @@ export function extractCritical(html) {
       ids[match[1]] = true
     }
   }
-
-  o.rules = sheet.sheet.slice().filter(x => {
-    RGX.lastIndex = 0
-    let match = RGX.exec(x)
-    const ret = match == null || ids[match[1]] || false
-    return ret
+  
+  o.rules = sheet.sheet.filter(x => {
+    const match = x.match(RGX)
+    return !match || ids[match[0]]
```

Comment:
Could you change `!match` to `match === null`, it's a bit more explicit and there will be a (extremely small and probably unimportant) performance improvement.